examples/rvv_strncpy.c

```diff
@@ -0,0 +1,66 @@
+#include "common.h"
+#include <riscv_vector.h>
+#include <string.h>
+
+// reference https://github.com/riscv/riscv-v-spec/blob/master/example/strncpy.s
+char *strncpy_vec(char *dst, char *src, size_t count) {
+  size_t vlmax = vsetvlmax_e8m1();
+  size_t vl;
+  long first_set_bit = -1;
+
+  while (first_set_bit < 0 && count > 0) {
+    vl = vsetvl_e8m1(count);
+
+    size_t new_vl;
+    vint8m1_t vec_src = vle8ff_v_i8m1(src, &new_vl, vl);
+
+    vbool8_t string_terminate = vmseq_vx_i8m1_b8(vec_src, 0, new_vl);
+    vbool8_t mask = vmsif_m_b8(string_terminate, new_vl);
+
+    vse8_v_i8m1_m(mask, dst, vec_src, new_vl);
+
+    count -= new_vl;
+    src += new_vl;
+    dst += new_vl;
+
+    first_set_bit = vfirst_m_b8(string_terminate, new_vl);
+  }
+
+  if (first_set_bit == -1)
```

Comment:
This is an extra branching instruction compared to the assembly implementation. Unlike assembly we can't insert a return if the two branching instructions don't perform a jump. (We also prefer not to use `goto`)

So an extra instruction is needed here. The extra logic "first_set_bit == -1` implies that the previous loop exited because count is reduced to zero.

Reply:
Thanks to @zakk, adjusted code and the extra logic is not needed.